### ISSUE TITLE
minsdk error when generating app descriptor

### ISSUE THREAD

**Issue by Kelsonk** [distriqt/ANE-GooglePlayServices#31]
Updated all ANE's today and got this error when trying to generate a new app descriptor

uses-sdk:minSdkVersion 21 cannot be smaller than version 24 declared in library C:\Users\kelso\OneDrive\Desktop\castleclickNew\apm\apm_packages\com.distriqt.playservices.Base\contents\platforms\android\AndroidManifest.xml as the library might be using APIs not available in 21
        Suggestion: use a compatible library with a minSdk of at most 21,
                or increase this project's minSdk version to at least 24,
                or use tools:overrideLibrary="com.distriqt.extension.playservices.base" to force usage (may lead to runtime failures)

**Comment by marchbold** [comment]
You need to add a custom android config. We are looking into an alternative way to handle this in apm. 

```
apm generate config android
```

Then edit `config/android/AndroidManifest.xml` and set the minimum sdk version to 24.

```xml
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android" xmlns:tools="http://schemas.android.com/tools">

    <uses-sdk android:minSdkVersion="24" android:targetSdkVersion="34" />

    <application>

        <meta-data android:name="android.max_aspect" android:value="2.5" />
        <meta-data android:name="android.notch_support" android:value="true"/>

    </application>

</manifest>
```

**Comment by marchbold** [comment]
Yeah this is an issue in apm, we are looking into it. Currently if you don't set a minimum version it sets it to 21 which is failing now that some of the libs have a dependency higher than that.

**Comment by Kelsonk** [comment]
That fixed it, in that it builds an application apm, and an iOS build currently works.

When doing an Android build, I get these errors.

applicationapm.xml: error 402: Current Android sdk version doesn't support property as xml tag
applicationapm.xml: error 402: Current Android sdk version doesn't support property as xml tag

Corresponding to these lines:
<property android:name="android.adservices.AD_SERVICES_CONFIG" android:resource="@xml/gma_ad_services_config"/>
<property android:name="android.adservices.AD_SERVICES_CONFIG" android:resource="@xml/ga_ad_services_config"/>

**Comment by mrfrasier** [comment]
@Kelsonk that's what brought me here. You hae to update the AIR SDK as well: https://github.com/airsdk/Adobe-Runtime-Support/discussions/1699#discussioncomment-9174571

AIR SDK 50.2.5.1 fixes this:
`Github-2976:Adding 'property' as a supported Android Manifest tag`

**Comment by Kelsonk** [comment]
The hits keep coming.

java.io.IOException: Unable to run java: com.adobe.air.ADTException: gradle tool failed:
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkReleaseDuplicateClasses'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.CheckDuplicatesRunnable
   > Duplicate class okio.AsyncTimeout found in modules jetified-okio-1.17.5 (okio-1.17.5.jar) and jetified-okio-jvm-3.4.0 (okio-jvm-3.4.0.jar)

Followed by a huge number of duplicate class jetified libraries.

**Comment by mrfrasier** [comment]
My issues ended there haha, so no direct help, but what I'd try next is just running apm update and make sure all the ANEs are in sync

**Comment by marchbold** [comment]
@Kelsonk Duplicate classes likely mean you have an old version of an extension or are using one we haven't updated yet. Could you post your project.apm or the list of extensions you are using?

@mrfrasier Is your build working now?

**Comment by Kelsonk** [comment]
> @Kelsonk Duplicate classes likely mean you have an old version of an extension or are using one we haven't updated yet. Could you post your project.apm or the list of extensions you are using?
> 
> @mrfrasier Is your build working now?

apm list shows this
├──androidx.cardview@1.0.6
├──com.distriqt.admob.AdColony@4.9.6
├──com.distriqt.admob.AppLovin@12.4.200
├──com.distriqt.admob.UnityAds@4.10.0
├──com.distriqt.Adverts@15.1.0
├──com.distriqt.Application@7.1.0
├──com.distriqt.Dialog@8.8.0
├──com.distriqt.Firebase@9.0.0
├──com.distriqt.firebase.Crashlytics@9.0.0
├──com.distriqt.firebase.Performance@9.0.0
├──com.distriqt.IDFA@5.2.0
├──com.distriqt.InAppBilling-Amazon@15.4.0
├──com.distriqt.Notifications@6.7.0
├──com.distriqt.playservices.Analytics@18.0.303
└──com.google.android.material@1.6.4

**Comment by mrfrasier** [comment]
> @Kelsonk Duplicate classes likely mean you have an old version of an extension or are using one we haven't updated yet. Could you post your project.apm or the list of extensions you are using?
> 
> @mrfrasier Is your build working now?

Yup, my build is good

**Comment by marchbold** [comment]
Thanks @Kelsonk. We are in process of updating the admob mediators, I believe your conflict will be from the `com.distriqt.admob.UnityAds` mediator. I'll make this a priority and get an update out shortly.
